WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/GiftCardInputViewModel.swift

```diff
@@ -24,3 +27,35 @@ final class GiftCardInputViewModel: ObservableObject {
         dismiss()
     }
 }
+
+private extension GiftCardInputViewModel {
+    func observeCodeForValidCheck() {
+        $code.removeDuplicates()
+            .map { self.isCodeValid($0) }
+            .assign(to: &$isValid)
+
+        $isValid.combineLatest($code)
+            .map { isValid, code in
+                isValid || code.isEmpty ? nil: Localization.errorMessage
+            }
+            .assign(to: &$errorMessage)
+    }
+
+    func isCodeValid(_ code: String) -> Bool {
+        let format = "^[A-Z0-9]{4}-[A-Z0-9]{4}-[A-Z0-9]{4}-[A-Z0-9]{4}$"
+
+        let regex = try? NSRegularExpression(pattern: format, options: .caseInsensitive)
```

Comment:
FYI: I think we can not use the [new Regex types](https://www.hackingwithswift.com/swift/5.7/regexes) but no need to change!

Reply:
TIL, this looks nice! However, when I tried it [it's iOS 16.0+](https://developer.apple.com/documentation/swift/regex) and we can't use it right now 😐